Client/game_sa/CDummyPoolSA.cpp

```diff
@@ -13,63 +13,93 @@
+    for (size_t i = 0; i < MAX_DUMMIES_DEFAULT; i++)
     {
-        CEntitySAInterface* object = (*m_ppDummyPoolInterface)->GetObject(i);
-        (*m_pLodBackup)[i] = object->m_pLod;
-        object->m_pLod = nullptr;
+        if (pDummyPool->IsContains(i))
+        {
+            CEntitySAInterface* building = pDummyPool->GetObject(i);
+
+            pGame->GetWorld()->Remove(building, CDummyPool_Destructor);
+            building->RemoveRWObjectWithReferencesCleanup();
+
+            pDummyPool->Release(i);
+
+            (*m_pOriginalElementsBackup)[i].first = true;
+            (*m_pOriginalElementsBackup)[i].second = *building;
+        }
+        else
+        {
+            (*m_pOriginalElementsBackup)[i].first = false;
+        }
     }
 }
 
-void CDummyPoolSA::RestoreAllBuildingsLods()
+void CDummyPoolSA::RestoreBackup()
 {
-    if (!m_pLodBackup)
+    if (!m_pOriginalElementsBackup)
         return;
 
-    for (int i = 0; i < MAX_DUMMIES; i++)
+    auto& originalData = *m_pOriginalElementsBackup;
+    auto  pDummyPool = (*m_ppDummyPoolInterface);
+    for (size_t i = 0; i < MAX_DUMMIES_DEFAULT; i++)
     {
-        CEntitySAInterface* object = (*m_ppDummyPoolInterface)->GetObject(i);
-        object->m_pLod = (*m_pLodBackup)[i];
+        if (originalData[i].first)
+        {
+            pDummyPool->AllocateAt(i);
+            auto pDummy = pDummyPool->GetObject(i);
+            *pDummy = originalData[i].second;
+
+            pGame->GetWorld()->Add(pDummy, CDummyPool_Constructor);
+        }
     }
 
-    m_pLodBackup.release();
+    m_pOriginalElementsBackup = nullptr;
 }
 
 void CDummyPoolSA::UpdateBuildingLods(void* oldPool, void* newPool)
 {
-    const uint32_t offset = (uint32_t)newPool - (uint32_t)oldPool;
+    const std::uint32_t offset = (std::uint32_t)newPool - (std::uint32_t)oldPool;
 
-    if (m_pLodBackup)
+    if (m_pOriginalElementsBackup)
     {
-        for (int i = 0; i < MAX_DUMMIES; i++)
+        for (int i = 0; i < MAX_DUMMIES_DEFAULT; i++)
         {
-            if ((*m_pLodBackup)[i] != nullptr)
+            if ((*m_pOriginalElementsBackup)[i].first)
             {
-                (*m_pLodBackup)[i] = (CEntitySAInterface*)((uint32_t)(*m_pLodBackup)[i] + offset);
+                CEntitySAInterface* object = &(*m_pOriginalElementsBackup)[i].second;
+                if (object->m_pLod)
+                {
+                    object->m_pLod = (CEntitySAInterface*)((std::uint32_t)(object->m_pLod) + offset);
+                }
             }
         }
     }
     else
     {
-        for (int i = 0; i < MAX_DUMMIES; i++)
+        for (int i = 0; i < MAX_DUMMIES_DEFAULT; i++)
         {
             CEntitySAInterface* object = (*m_ppDummyPoolInterface)->GetObject(i);
             if (object->m_pLod)
             {
-                object->m_pLod = (CEntitySAInterface*)((uint32_t)object->m_pLod + offset);
+                object->m_pLod = (CEntitySAInterface*)((std::uint32_t)object->m_pLod + offset);
```

Comment:
Early continue

Reply:
This is one line. It's totally unnecessary in this case